charts/nri-kube-events/templates/_helpers_compatibility.tpl

```diff
@@ -3,8 +3,144 @@ Returns a dictionary with legacy runAsUser config.
 We know that it only has "one line" but it is separated from the rest of the helpers because it is a temporary things
 that we should EOL. The EOL time of this will be marked when we GA the deprecation of Helm v2.
 */}}
-{{- define "newrelic.compatibility.securityContext.pod" -}}
-{{- if  .Values.runAsUser -}}
+{{- define "nri-kube-events.compatibility.securityContext.pod" -}}
+{{- if .Values.runAsUser -}}
 runAsUser: {{ .Values.runAsUser }}
 {{- end -}}
 {{- end -}}
+
+
+
+{{/*
+Creates the image string needed to pull the integration image respecting the breaking change we made in the values file
+*/}}
+{{- define "nri-kube-events.compatibility.images.integration" -}}
+{{- /* workaround: https://github.com/helm/helm/issues/9266 */ -}}
+{{- $values := (.Values | merge (dict)) -}}
+
+{{- $oldRegistry := dig "image" "kubeEvents" "registry" "" $values }}
+{{- $newRegistry := dig "images" "integration" "registry" "" $values }}
+{{- $registry := $oldRegistry | default $newRegistry }}
+
+{{- $oldRepository := dig "image" "kubeEvents" "repository" "" $values }}
+{{- $newRepository := dig "images" "integration" "repository" "" $values }}
+{{- $repository := $oldRepository | default $newRepository }}
+
+{{- $oldTag := dig "image" "kubeEvents" "tag" "" $values }}
+{{- $newTag := dig "images" "integration" "tag" "" $values }}
+{{- $tag := $oldTag | default $newTag }}
+
+{{- if $registry -}}
+    {{- printf "%s/%s:%s" $registry $repository $tag -}}
+{{- else -}}
+    {{- printf "%s:%s" $repository $tag -}}
```

Comment:
once we discover how the "image root" is called could we use the common library to render the image field?

Otherwise globals are not taken into account. Or I am missing something?

Reply:
You are right...

Reply:
In fact, we are not merging `imagePullSecrets` which could be another good approach. I am redoing it.

Reply:
The conclusion to this is "do to create a PR at 18 in the afternoon"

Reply:
I missed the globals thing too 😓 , thanks for pointing it out @paologallinaharbur

Reply:
Solved in [#9a19c07](https://github.com/newrelic/nri-kube-events/pull/117/commits/9a19c070d4e2903356e9be3eb00fc8e47c103908)